packages/server-common/src/handlers/calendar/calendar-handler-factory.contentful.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  EntryFieldTypes,
+  getCDAClient,
+  pollContentfulDeliveryApi,
+} from '@asap-hub/contentful';
+import { CalendarDataProvider, CalendarEvent, gp2 } from '@asap-hub/model';
+import { EventBridgeEvent } from 'aws-lambda';
+import 'source-map-support/register';
+
+import { Alerts, Logger } from '../../utils';
+import { validateBody } from '../../validation/calendar-handler.contentful.validation';
+import { CalendarContentfulPayload } from '../event-bus';
+import {
+  SubscribeToEventChanges,
+  UnsubscribeFromEventChanges,
+} from './subscribe-handler';
+
+type ContentfulDeliveryApiConfig = {
+  space: string;
+  environment: string;
+  accessToken: string;
+};
+
+type CalendarSkeleton = {
+  contentTypeId: 'calendars';
+  fields: {
+    googleApiMetadata: EntryFieldTypes.Object<{
+      associatedGoogleCalendarId?: string | null;
+      resourceId?: string | null;
+    }>;
+  };
+};
+
+export const calendarCreatedContentfulHandlerFactory =
+  (
+    subscribe: SubscribeToEventChanges,
+    unsubscribe: UnsubscribeFromEventChanges,
+    calendarDataProvider: CalendarDataProvider | gp2.CalendarDataProvider,
+    alerts: Alerts,
+    logger: Logger,
+    contentfulDeliveryApiConfig: ContentfulDeliveryApiConfig,
+  ) =>
+  async (
+    event: EventBridgeEvent<CalendarEvent, CalendarContentfulPayload>,
+  ): Promise<'OK'> => {
+    logger.debug(JSON.stringify(event, null, 2), 'Event input');
+
+    const eventType = event['detail-type'];
+    const {
+      fields,
+      sys,
+      resourceId: calendarId,
+    } = validateBody(event.detail as never);
+
+    const webhookEventVersion = sys.revision;
+    const webhookEventGoogleCalendarId = fields.googleCalendarId['en-US'];
+
+    logger.info(
+      `Received a '${eventType}' event for the calendar ${calendarId}`,
+    );
+
+    logger.debug(`Event payload: ${JSON.stringify(fields)}`);
+
+    if (eventType !== 'CalendarsPublished') return 'OK';
+
+    const cdaClient = getCDAClient(contentfulDeliveryApiConfig);
+
+    const fetchCalendarById = () =>
+      cdaClient.getEntry<CalendarSkeleton>(calendarId);
+
+    let cmsCalendar;
+    try {
+      cmsCalendar = await pollContentfulDeliveryApi<CalendarSkeleton>(
```

Comment:
Hmm I think there's a point because we came to the conclusion that this needs to check the last version in case we have something like this below:

<img width="1218" alt="Screenshot 2023-05-08 at 06 54 01" src="https://github.com/yldio/asap-hub/assets/16595804/d3048aea-a10f-49f4-bacc-192848235ba8">

and if the webhook thing is a nice to have, it may cause some bugs while not implemented